include/spl/sys/timer.h

```diff
@@ -72,4 +72,30 @@ usleep_range(unsigned long min, unsigned long max)
 #define	USEC_TO_TICK(us)		usecs_to_jiffies(us)
 #define	NSEC_TO_TICK(ns)		usecs_to_jiffies(ns / NSEC_PER_USEC)
 
+#ifndef from_timer
+#define	from_timer(var, timer, timer_field) \
+	container_of(timer, typeof(*var), timer_field)
+#endif
+
+
+#ifdef HAVE_KERNEL_TIMER_FUNCTION_TIMER_LIST
+#define	TIMER_FUNC_ARG	struct timer_list *
+#else
+#define	TIMER_FUNC_ARG	unsigned long
+#endif
```

Comment:
Rather then define this as a macro, can  we add our own typedef for this?  Or does it need to be a macro for the `timer_setup()` prototype below.  

```c
#if defined(HAVE_KERNEL_TIMER_FUNCTION_TIMER_LIST)
typedef struct timer_list *     spl_timer_list_t;
#else
typedef unsigned long           spl_timer_list_t;
#endif
```